standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnStore.java

```diff
@@ -522,6 +522,16 @@ void onRename(String oldCatName, String oldDbName, String oldTabName, String old
    */
   void markRefused(CompactionInfo info) throws MetaException;
 
+  /**
+   * Updates the cleaner retry time related information (compaction properties and commit time) of the CompactionInfo
+   * in the HMS database.
+   * @param info The {@link CompactionInfo} object to be updated.
+   * @param retentionTime The time until the clean won't be attempted again.
+   * @throws MetaException
+   */
+  @RetrySemantics.CannotRetry
+  void updateCompactionRetry(CompactionInfo info, long retentionTime) throws MetaException;
```

Comment:
method name doesn't conform to its declaration. should be something like retryCleanerAttemptWithBackoff()